ledger-core/virtual/integration/method_test.go

```diff
@@ -1704,3 +1726,194 @@ func TestVirtual_CallContractTwoTimes(t *testing.T) {
+		outgoingCallRef = reference.NewRecordOf(objectBGlobal, callOutgoing)
+	)
+
+	Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+	Method_PrepareObject(ctx, server, payload.Ready, objectBGlobal)
+
+	// add ExecutionMocks to runnerMock
+	{
+		outgoingCall := execution.NewRPCBuilder(outgoingCallRef, objectBGlobal).
+			CallMethod(objectAGlobal, classA, "Bar", []byte("123")).
+			SetInterference(barIsolation.Interference).
+			SetIsolation(barIsolation.State)
+		objectBExecutionMock := runnerMock.AddExecutionMock("Foo")
+		objectBExecutionMock.AddStart(
+			func(ctx execution.Context) {
+				logger.Debug("ExecutionStart [B.Foo]")
+				assert.Equal(t, objectBGlobal, ctx.Object)
+				assert.Equal(t, callOutgoing, ctx.Request.CallOutgoing)
+			},
+			&execution.Update{
+				Type:     execution.OutgoingCall,
+				Error:    nil,
+				Outgoing: outgoingCall,
+			},
+		)
+		objectBExecutionMock.AddContinue(
+			func(result []byte) {
+				logger.Debug("ExecutionContinue [B.Foo]")
+				require.Equal(t, []byte("finish A.Bar"), result)
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("finish B.Foo"), objectBGlobal),
+			},
+		)
+
+		runnerMock.AddExecutionMock("Bar").AddStart(
+			func(ctx execution.Context) {
+				logger.Debug("ExecutionStart [A.Bar]")
+				assert.Equal(t, objectAGlobal, ctx.Request.Callee)
+				assert.Equal(t, objectBGlobal, ctx.Request.Caller)
+				assert.Equal(t, []byte("123"), ctx.Request.Arguments)
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("finish A.Bar"), objectBGlobal),
+			},
+		)
+
+		runnerMock.AddExecutionClassify("Bar", barIsolation, nil)
+		runnerMock.AddExecutionClassify("Foo", fooIsolation, nil)
+	}
+
+	callCount := 0
+
+	firstPulse := server.GetPulse().PulseNumber
+
+	{
+		typedChecker.VStateReport.SetResend(true)
+		typedChecker.VDelegatedCallRequest.SetResend(true)
+		typedChecker.VDelegatedCallResponse.SetResend(true)
+		typedChecker.VDelegatedRequestFinished.SetResend(true)
+		typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
+			assert.Equal(t, objectAGlobal, request.Callee)
+			assert.Equal(t, objectBGlobal, request.Caller)
+			assert.Equal(t, []byte("123"), request.Arguments)
+			assert.Equal(t, payload.CTMethod, request.CallType)
+			assert.Equal(t, uint32(1), request.CallSequence)
+			assert.Equal(t, outgoingCallRef, request.CallReason)
+			assert.Equal(t, payload.BuildCallFlags(barIsolation.Interference, barIsolation.State), request.CallFlags)
+
+			assert.Equal(t, firstPulse, request.CallOutgoing.Pulse())
+
+			if callCount == 0 {
+				assert.Equal(t, payload.CallDelegationToken{}, request.DelegationSpec)
+				server.IncrementPulseAndWaitIdle(ctx)
+				callCount++
+				// request will be sent in previous pulse
+				// ommit sending
+				return false
+			}
+
+			service := authentication.NewService(ctx, server.JetCoordinatorMock)
+			origToken := service.GetCallDelegationToken(
```

Comment:
No-no, you don't use platform code for checks. You should check every field using test constants.

Reply:
could you explain why? Result of GetCallDelegationToken output should be tested elsewhere. If GetCallDelegationToken logic will change, why do you want this test to fail?